exporter/sentryexporter/sentry_exporter.go

```diff
@@ -42,3 +45,172 @@ func CreateSentryExporter(config *Config) (component.TraceExporter, error) {
+// https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions
+func generateSpanDescriptors(name string, attrs pdata.AttributeMap, spanKind pdata.SpanKind) (op string, description string) {
+	var opString strings.Builder
+	var dString strings.Builder
+
+	// if http.method exists, this is an http request span
+	if httpMethod, ok := attrs.Get(conventions.AttributeHTTPMethod); ok {
+		opString.WriteString("http")
+
+		switch spanKind {
+		case pdata.SpanKindCLIENT:
+			opString.WriteString(".client")
+		case pdata.SpanKindSERVER:
+			opString.WriteString(".server")
+		}
+
+		// Ex. description="GET /api/users/{user_id}"
+		dString.WriteString(httpMethod.StringVal())
+		dString.WriteString(" ")
+		dString.WriteString(name)
+
+		return opString.String(), dString.String()
+	}
+
+	// if db.type exists then this is a database call span
+	if _, ok := attrs.Get(conventions.AttributeDBType); ok {
+		// TODO: Use more detailed op code?
+		opString.WriteString("db")
+
+		// Use DB statement (Ex "SELECT * FROM table") if possible as description
+		if statement, okInst := attrs.Get(conventions.AttributeDBStatement); okInst {
+			dString.WriteString(statement.StringVal())
+		} else {
+			dString.WriteString(name)
+		}
+
+		return opString.String(), dString.String()
+	}
+
+	// if rpc.service exists then this is a rpc call span
+	if _, ok := attrs.Get(conventions.AttributeRPCService); ok {
+		opString.WriteString("rpc")
+
+		return opString.String(), name
+	}
+
+	// if messaging.system exists then this is a messaging system span
+	if _, ok := attrs.Get("messaging.system"); ok {
+		opString.WriteString("message")
+
+		return opString.String(), name
+	}
+
+	// if faas.trigger exists then this is a function as a service span
+	if trigger, ok := attrs.Get("faas.trigger"); ok {
+		opString.WriteString(trigger.StringVal())
+
+		return opString.String(), name
+	}
+
+	// Default just use span.name
+	return "", name
+}
+
+func generateTagsFromAttributes(attrs pdata.AttributeMap) Tags {
+	tags := make(map[string]string)
+
+	attrs.ForEach(func(key string, attr pdata.AttributeValue) {
+		switch attr.Type() {
+		case pdata.AttributeValueSTRING:
+			tags[key] = attr.StringVal()
+		case pdata.AttributeValueBOOL:
+			tags[key] = strconv.FormatBool(attr.BoolVal())
+		case pdata.AttributeValueDOUBLE:
+			tags[key] = strconv.FormatFloat(attr.DoubleVal(), 'g', -1, 64)
+		case pdata.AttributeValueINT:
+			tags[key] = strconv.FormatInt(attr.IntVal(), 10)
+		}
+	})
+
+	return tags
+}
+
+var canonicalCodes = [...]string{
+	"ok",
+	"cancelled",
+	"unknown",
+	"invalid_argument",
+	"deadline_exceeded",
+	"not_found",
+	"already_exists",
+	"permission_denied",
+	"resource_exhausted",
+	"failed_precondition",
+	"aborted",
+	"out_of_range",
+	"unimplemented",
+	"internal",
+	"unavailable",
+	"data_loss",
+	"unauthenticated",
+}
+
+func generateStatusFromSpanStatus(status pdata.SpanStatus) (string, string) {
```

Comment:
When a function returns values that have unclear meaning, it is customary to [use named return values to document that they are](https://golang.org/doc/effective_go.html#named-results).

```suggestion
func statusFromSpanStatus(status pdata.SpanStatus) (status string, message string) {
```

In the spirit of [Effective Go: Getters](https://golang.org/doc/effective_go.html#Getters), I'd also suggest using a shorter func name, the `generate` prefix doesn't add as much to readability as it adds characters to make lines unnecessarily longer.

See the call site:

```go
status, message := statusFromSpanStatus(span.Status())
```